futures-util/src/try_future/mod.rs

```diff
@@ -501,4 +504,136 @@ pub trait TryFutureExt: TryFuture {
     {
         IntoFuture::new(self)
     }
+
+    /// Joins the result of two futures, waiting for them both to complete or
+    /// for one to produce an error.
+    ///
+    /// This function will return a new future which awaits both this and the
+    /// `other` future to complete. If successful, the returned future will
+    /// finish with a tuple of both results. If unsuccesful, it will complete
+    /// with an error.
+    ///
+    /// Note that this function consumes the receiving future and returns a
+    /// wrapped version of it.
```

Comment:
Note that this method consumes the futures it is called on and returns a wrapped version of them.

Reply:
The existing wording here is the one used elsewhere in the lib. IMO we should just remove this bit altogether, as I think it's obvious from the `self` parameter, but whatever we do should be consistent across the library.

Reply:
- The wording here needs to be in plural
- It's consistent with the other TryFuture combinators. I changed that a while back and still need to update some locations. I don't like the wording "receiving future"

Reply:
About removing it: I wouldn't be opposed. But, I suggest to keep it